examples/checkpointing/checkpoint.py

```diff
@@ -13,22 +13,28 @@ class CheckpointOperator(Operator):
                     automatically include these cached arguments in the call to the
                     underlying devito.Operator.apply().
     """
-    t_arg_names = {'t_start': 'time_s', 't_end': 'time_e'}
+    t_arg_names = {'t_start': 'time_m', 't_end': 'time_M'}
 
     def __init__(self, op, **kwargs):
         self.op = op
-        self.time_order = kwargs.pop('time_order', 1)
         self.args = kwargs
+        t = {self.t_arg_names['t_end']: 0}
+        op_default_args = self.op.arguments(**t)
+        self.start_offset = op_default_args[self.t_arg_names['t_start']]
```

Comment:
So this contains the time order now?

Reply:
It's not time order because I see some order 2 kernels starting from 1.

Reply:
@navjotk

Reply:
???

Reply:
tagged you before I saw your comment (I hadn't refreshed the window) -- ignore it